### ISSUE TITLE
[Feature Request] Stub DLL Configuration Update

### ISSUE THREAD

**Issue by SolsticeSpectrum** [Bas4c/SteamBypass#1]
I like this project for it's simplicity, it is more lightweight than Goldberg and I like such projects so this peaked my interest.
However I have some recommendations:

I feel like it's kinda redundant to have separate file just for AppId.
It should be moved over to the same file as interfaces.
A new property called `[General]` or `[Settings]` could be added and AppId move there.

If you choose to do this I also recommend adding more options.
Must have options are:

```txt
Username - Sets the Username
SteamId  - AccountId would be single integer which serves as "seed" for SteamId generation.
         - 0 would give random ID maybe by using some system info so it's always same on current device.
         - 1 to infinity would serve as seed.
         - SteamId would set the SteamID all together (useful for transfering save files when you choose to buy the game).
         - For example Elden Ring save file is bound to SteamID, you can't just move the save file, you have to hex edit it.
Language - This is inherited from system but should be optional as well if present.
Overlay  - Enable or disable Steam Overlay.
```

There should also be another property `[DLC]` which would hold DLC related settings.
Primarily that would be `UnlockAll` and list of such entries as `ID=Name` which would dictate the DLC names and IDs.

**Comment by Bas4c** [comment]
SteamId can't be random, due to in my testing with **Source Engine** games, it will crash when i use random SteamId. Reason it fails in game IsValid check, this is way i'm choose to use static SteamId and AppId in separate file is tactical decision, due to **Source Engine** games use it.

But i will implement your other suggested features, **[DLC]** stuff and **Language** selection, but still if not specified it will fallback to system language and if system language is not supported to "english".

About **Overlay** feature i don't have idea how to implement, due this will mean i need to create some sorts of dialog UI for that, please if you can provide me some UI design screenshots for reference.

I'm not able to test with resource intensive games my stub DLL, due to my PC doesn't have enough RAM and VRAM for this games. In short terms my PC is not gaming PC.

**Comment by SolsticeSpectrum** [comment]
I think the Overlay in **CODEX** just allows Steam Overlay to work when added as non-steam game to Steam. It enables the `Shift+Tab` overlay to work. How to is beyond me. Overlay to work. How to is beyond me.

For SteamId, could there at least be the option to set custom SteamId? Like for example get my real SteamId and supply it?

I tested your DLL with **Elden Ring Shadow of the Erdtree** and **Slime Rancher** and both work like charm. I'll test with more games and let you know if there are some that don't work. **Deadlock** doesn't work and Xbox doesn't work but those are unreleased games and Xbox alone has some custom verification method so no surprised. _**Deadlock** can work but you have to run the game with specific command line parameters._

**Comment by Bas4c** [comment]
i can add SteamId config, but i will add comment above in example INI file and by default i will comment out this config in example INI file, for everyone to know it's risky config "You can supply your own SteamId, but use it on your own risk".

After i implement this updates, i release it as new version "1.75", to preserve this version.

Also i never used SteamClient on my own, this why i only familiar with steam_api from reverse engineering it and public SteamworksSDK repository.

**Comment by SolsticeSpectrum** [comment]
Sounds good to me, I wish I could tell you how the overlay thing works. It's nice to have working overlay in non-steam games if you add that to your steam library.

Maybe this? Also it's experimental, probably unstable
https://github.com/metrixmedia/SteamEmulator/blob/main/overlay_experimental/steam_overlay.cpp

**Comment by Bas4c** [comment]
Now i will implement SteamId, Language, UserName configuration.

About Overlay this is good reference, **but i been glad if you design UI for me for that overlays not code only UI design image**, you can send it here as image of UI design, this will make faster development, but as i never use external libraries, i will implement it using Win32 API functions, to manipulate UI and this will make it more lightweight as hole this project point is to make it as lightweight as possible, no third party dependencies expect Win32 API.

Overley feature will be not so soon, maybe after month or two.

**Comment by SolsticeSpectrum** [comment]
I like the lightness, no pressure

**Comment by Bas4c** [comment]
Also main games i testing my stub DLL with on my own is **Source Engine**  games, due to this games is quite optimized for middle hardware and doesn't require to much RAM.

**Comment by Bas4c** [comment]
After update "steam_api.ini" will have this format, i used **(base 16)** for SteamId, due to it's faster to parse, then **(base 10)**.

```ini
[Username]

 ; 
 ; Your username
 ; 
 ; 128 character limit
 ; 

 Value=LocalPlayer

[Language]

 ; 
 ; SteamClient supported languages
 ; 
 ; arabic, bulgarian, schinese, tchinese, czech, danish,
 ; dutch, english, finnish, french, german, greek,
 ; hungarian, indonesian, italian, japanese, koreana, norwegian,
 ; polish, brazilian, portuguese, romanian, russian, latam,
 ; spanish, swedish, thai, turkish,
 ; ukrainian, vietnamese
 ; 

 Value=english

[SteamId]

 ; 
 ; You can supply your own SteamId64 in (base 16), but use it on your own risk
 ; 
 ; eUniverse: 0xFF00000000000000
 ; eAccountType: 0x00F0000000000000
 ; ObjectId: 0x000FFFFF00000000
 ; Instance: 0x00000000FFFFFFFF
 ; 

 Value=0110000100000001

[SteamAPI]

 SteamAppList=STEAMAPPLIST_INTERFACE_VERSION001
 SteamApps=STEAMAPPS_INTERFACE_VERSION007
 SteamClient=SteamClient017
 SteamController=SteamController003
 SteamFriends=SteamFriends015
 SteamGameServer=SteamGameServer012
 SteamGameServerStats=SteamGameServerStats001
 SteamHTMLSurface=STEAMHTMLSURFACE_INTERFACE_VERSION_003
 SteamHTTP=STEAMHTTP_INTERFACE_VERSION002
 SteamInput=SteamInput006
 SteamInventory=STEAMINVENTORY_INTERFACE_V001
 SteamMatchmaking=SteamMatchMaking009
 SteamMatchmakingServers=SteamMatchMakingServers002
 SteamMusicRemote=STEAMMUSICREMOTE_INTERFACE_VERSION001
 SteamGameSearch=SteamMatchGameSearch001
 SteamParties=SteamParties002
 SteamMusic=STEAMMUSIC_INTERFACE_VERSION001
 SteamNetworking=SteamNetworking005
 SteamNetworkingMessages=SteamNetworkingMessages002
 SteamNetworkingSockets=SteamNetworkingSockets008
 SteamNetworkingUtils=SteamNetworkingUtils002
 SteamParentalSettings=STEAMPARENTALSETTINGS_INTERFACE_VERSION001
 SteamRemotePlay=STEAMREMOTEPLAY_INTERFACE_VERSION001
 SteamRemoteStorage=STEAMREMOTESTORAGE_INTERFACE_VERSION013
 SteamScreenshots=STEAMSCREENSHOTS_INTERFACE_VERSION002
 SteamUGC=STEAMUGC_INTERFACE_VERSION007
 SteamUnifiedMessages=STEAMUNIFIEDMESSAGES_INTERFACE_VERSION001
 SteamUser=SteamUser018
 SteamUserStats=STEAMUSERSTATS_INTERFACE_VERSION011
 SteamUtils=SteamUtils007
 SteamVideo=STEAMVIDEO_INTERFACE_V001
 SteamMasterServerUpdater=SteamMasterServerUpdater001
```

**Comment by Bas4c** [comment]
This update is ready and can be downloaded here: [FreeAPI 1.75](https://github.com/Bas4c/SteamBypass/releases/tag/FreeAPI.v1.75).
I'm closing this **[Feature Request]**, due to i'm not sure if i will implement **Overlay** stuff at all, but i implemented other suggested features.

**Comment by SolsticeSpectrum** [comment]
@Bas4c What about the DLC stuff? Also how does one convert his SteamID to Base16? This ID 0110000100000001 corresponds to this SteamID 76561197960265729.

Nvm, it's the HEX value
![image](https://github.com/user-attachments/assets/20505f70-97da-44bd-92d3-96645c875e37)

Maybe in the ini, say that it's the hex value of SteamID64, because it's more clear than saying base16

**Comment by SolsticeSpectrum** [comment]
![image](https://github.com/user-attachments/assets/30c25aae-dbf4-4882-aea2-d5268c31a3ae)
Maybe initialize g_GameAppId with -1? User might set it to 0 which will make this run everytime, app id is requested

**Comment by Bas4c** [comment]
About AppId i can't set to -1, due to it is Unsigned 32-bit integer and "Real" game AppId's never is 0, this is way it's fine.

About SteamId "0110000100000001" i used as default, due to it's tested SteamId by me with **Source Engine** games, that works, but if you decide to change it you must comply to original SteamworksSDK **CSteamID** IsValid check, to make sure it's valid for game.

**Comment by SolsticeSpectrum** [comment]
I see, thanks for clarification. What about the DLCs?

**Comment by Bas4c** [comment]
I'm also glad you started this issue, not only due to your good suggestions, but also due to i been having someone to talk to, as i don't have any real friends, to talk to. I thank you so much.

**Comment by SolsticeSpectrum** [comment]
@Bas4c If you want, you can hit me up on Discord, I can tell you how I stumbled on this repository, just add me under `thed*m*dude`

**Comment by SolsticeSpectrum** [comment]
@Bas4c I just noticed, it censored the name, it's `t h e d u m b d u d e` without spaces

**Comment by Bas4c** [comment]
@SolsticeSpectrum fixed in **FreeAPI v1.75** ISteamHTMLSurface vtable method order, you should download updated archive, due to before update it contains incorrect vtable method order inside ISteamHTMLSurface vtable: [FreeAPI v1.75](https://github.com/Bas4c/SteamBypass/releases/tag/FreeAPI.v1.75)
